cmd/dashboard.go

```diff
@@ -76,19 +83,81 @@ var dashboardCmd = &cobra.Command{
 }
 
 func runPortforward(podName string, localPort int) error {
-	cmd, err := exec.LookPath("kubectl")
+	stopChannel := make(chan struct{}, 1)
+	readyChannel := make(chan struct{})
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+	defer signal.Stop(signals)
+
+	go func() {
+		<-signals
+		if stopChannel != nil {
+			close(stopChannel)
+		}
+	}()
+
+	// Open the Dashboard in a browser when the port-forward is established
+	go func() {
+		<-readyChannel
+		openInBrowser(fmt.Sprintf("http://localhost:%d", localPort))
+	}()
+
+	fw, err := newPortforwarder(podName, localPort, stopChannel, readyChannel)
 	if err != nil {
 		return err
 	}
-	args := []string{"kubectl", "--namespace", ingressNamespace, "port-forward", podName, fmt.Sprintf("%d:80", localPort)}
+	return fw.ForwardPorts()
+}
 
-	env := os.Environ()
+func newPortforwarder(podName string, localPort int, stopChannel, readyChannel chan struct{}) (*portforward.PortForwarder, error) {
+	config, err := restClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := portforwardReqURL(config, podName)
+	if err != nil {
+		return nil, err
+	}
+
+	dialer, err := remotecommand.NewExecutor(config, "POST", url)
+	if err != nil {
+		return nil, err
+	}
+
+	ports := []string{fmt.Sprintf("%d:80", localPort)}
+
+	return portforward.New(dialer, ports, stopChannel, readyChannel, os.Stdout, os.Stderr)
+}
+
+func restClientConfig() (*rest.Config, error) {
```

Comment:
you should probably move this function to the `root` file

Reply:
This is a very specific client config just for building the port-forward URI, so I think it should stay here.

Reply:
mm, I saw it generic, what is the specific part? maybe the `config.NegotiatedSerializer`? If that is the case I would specify somewhere that this restClient is specific for doing port-forward (apart from defining that in this file) so other people don't misuse it. (Since this function is available for the whole `cmd` package)

Reply:
It's not generic, we're specifying the api path and group version so we get URLs prefixed with `api/v1`. I don't think we need to use this anywhere else do we? We should typically use the discovery API.

Reply:
I think this looks like a regular v1 config.  Can we just use `clientset.CoreV1().RESTClient()` (or whatever the correct function is) below?

Reply:
Ah, so I wanted to, since this is actually what `kubectl` uses also, but @ngtuna suggested not using it in https://github.com/kubeapps/kubeapps/pull/41#discussion_r151346356 to reduce the dependency. However, now that I think about it, I don't think this would be pulling in many more dependencies, it uses the same underlying stuff. @ngtuna what do you think about using clientset for both this and checking if the nginx-ingress pod is there too?

Reply:
Yah I am seeing no problem now for using `clientset` as we generally used it in several functions. What I was saying meant that if we can do stuffs easily with the current-in-used `resourceClient` then we should keep using it